File: Oats/Quake3.js
function Q_rsqrt(number) {
  var x2, y;
  var i;

  x2 = number * 0.5;
  y = number;
  i = new Int32Array(new Float32Array([y]).buffer)[0];;
  i = 0x5f3759df - (i >> 1);
  y = new Float32Array(new Int32Array([i]).buffer)[0];;

  y = y*(1.5 - (x2 * y * y));
  y = y*(1.5 - (x2 * y * y));

  return y;
}

console.log(Q_rsqrt(0.15265));

File: Oats/Quake3.java
public class Main {
  public static void main(String[] args) {
    float number = 16.0F; 
    float result = Q_rsqrt(number);
    System.out.println("Result: " + result);
  }

  public static float Q_rsqrt(float number) {
    int i;
    float x2, y;
    final float threeHalfs = 1.5F;
    
    x2 = number * 0.5F;
    y = number;
    i = Float.floatToIntBits(y); 
    i = 0x5f3759df - (i >> 1); 
    y = Float.intBitsToFloat(i);

    y = y * (threeHalfs - (x2 * y * y));
    // Uncomment the following line for the second iteration
    // y = y * (threeHalfs - (x2 * y * y)); // Second iteration
    return y;
  }
}
